### ISSUE TITLE
slowness and rendering bugs with long lines

### ISSUE THREAD

**Issue by sp3d** [eyelash/xi-gtk#19]
Create a file with long lines, e.g. by running:
$ `echo -ne "as\nbas\nasdgsdga\nbhfsd\nas\neq4515g\nafxdvsfvbzxcb xcv\nok\n\n\n" > test.txt; for i in `seq 15`; do paste -d'x' test.txt test.txt > test2.txt; mv test2.txt test.txt; done`

Open it in xi-gtk, and all interactions are very slow; in addition, line 7 appears to have multiple overlapping pieces of text rendered.

**Comment by winks** [comment]
I have the exact same problem with `gedit 3.22.0-2` on Debian stretch, that file looks identical and broken.

**Comment by eyelash** [comment]
This might actually be a problem in Pango.

**Comment by sp3d** [comment]
Yes (it smells kind of like an integer overflow), but we simply shouldn't be rendering more of a line than will fit onscreen.
